src/credentials/PersistentStorageOpCertStore.cpp

```diff
@@ -319,6 +386,97 @@ CHIP_ERROR PersistentStorageOpCertStore::UpdateOpCertsForFabric(FabricIndex fabr
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR PersistentStorageOpCertStore::BasicVidVerificationAssumptionsAreMet(FabricIndex fabricIndex) const
+{
+    VerifyOrReturnError(mStorage != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+    // Must already have a valid NOC chain.
+    VerifyOrReturnError(HasNocChainForFabric(fabricIndex), CHIP_ERROR_INCORRECT_STATE);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOpCertStore::UpdateVidVerificationSignerCertForFabric(FabricIndex fabricIndex, ByteSpan vvsc)
+{
+    ReturnErrorOnFailure(BasicVidVerificationAssumptionsAreMet(fabricIndex));
+    VerifyOrReturnError(vvsc.empty() || vvsc.size() <= Credentials::kMaxCHIPCertLength, CHIP_ERROR_INVALID_ARGUMENT);
+
+    // Can't try to set a VVSC if ICAC present.
+    bool legalVvscSetting = vvsc.empty() || !HasCertificateForFabric(fabricIndex, CertChainElement::kIcac);
+    VerifyOrReturnError(legalVvscSetting, CHIP_ERROR_INCORRECT_STATE);
+
+    CHIP_ERROR vvscErr = CHIP_NO_ERROR;
+
+    if (vvsc.empty())
+    {
+        if (fabricIndex == mPendingFabricIndex)
+        {
+            mPendingVvsc.Free();
+            mStateFlags.Set(StateFlags::kVvscUpdated);
+        }
+        else
+        {
+            vvscErr = DeleteVidVerificationElementFromStorage(mStorage, fabricIndex, VidVerificationElement::kVvsc);
+        }
+    }
+    else
+    {
+        if (fabricIndex == mPendingFabricIndex)
+        {
+            VerifyOrReturnError(mPendingVvsc.Alloc(vvsc.size()), CHIP_ERROR_NO_MEMORY);
+            memcpy(mPendingVvsc.Get(), vvsc.data(), vvsc.size());
+            mStateFlags.Set(StateFlags::kVvscUpdated);
+        }
+        else
+        {
+            vvscErr = SaveVidVerificationElementToStorage(mStorage, fabricIndex, VidVerificationElement::kVvsc, vvsc);
+        }
+    }
+
+    return vvscErr;
+}
+
+CHIP_ERROR PersistentStorageOpCertStore::UpdateVidVerificationStatementForFabric(FabricIndex fabricIndex,
+                                                                                 ByteSpan vidVerificationStatement)
+{
+    ReturnErrorOnFailure(BasicVidVerificationAssumptionsAreMet(fabricIndex));
+    VerifyOrReturnError(vidVerificationStatement.empty() ||
```

Comment:
Again, it would be nice if this size check were just in the cluster...

Reply:
I can do that for sure. I was trying to cover more unit tests in the opcerstore since the cluster is not unit testable yet, due to legacy design, but it has an integration test that may suffice.